File: 6kyu 1/spinWords.js
function spinWords(str) {
   const arr = str.split(" ");
   let result = [];
   for (let i in arr) {
      if (arr[i].length >= 5) {
         arr[i] = arr[i].split("").reverse().join("");
         result.push(arr[i]);
      } else {
         result.push(arr[i]);
      }
   }
   return result.join(" ");
}


function spinWords(words) {
   return words.split(' ').map(function (word) {
      return (word.length > 4) ? word.split('').reverse().join('') : word;
   }).join(' ');
}


function spinWords(str) {
   return str.split(' ').map(w => w.length < 5 ? w : w.split('').reverse().join('')).join(' ');
}


function spinWords(string) {
   return string.replace(/\w{5,}/g, function (w) {
      return w.split('').reverse().join('')
   })
}


console.log(spinWords("Welcome"), "emocleW");
console.log(spinWords("Hey fellow warriors"), "Hey wollef sroirraw");
console.log(spinWords("This is a test"), "This is a test");
console.log(spinWords("This is another test"), "This is rehtona test");
console.log(spinWords("You are almost to the last test"), "You are tsomla to the last test");
console.log(spinWords("Just kidding there is still one more"), "Just gniddik ereht is llits one more");
console.log(spinWords("Seriously this is the last one"), "ylsuoireS this is the last one");

File: Python/spin_words.py
"""
|--------------------------------------------------------------------------
| Stop gninnipS My sdroW!
|--------------------------------------------------------------------------
|
"""

def spin_words(sentence):
   arr = sentence.split(" ")
   res = []
   for i in range(len(arr)):
      if len(arr[i]) >= 5:
         res.append(''.join(reversed(arr[i])))
      else:
         res.append(arr[i])
   return ' '.join(res)


def spin_words(sentence):
   return " ".join([x[::-1] if len(x) >= 5 else x for x in sentence.split(" ")])


import re
def spin_words(sentence):
   return re.sub(r"\w{5,}", lambda w: w.group(0)[::-1], sentence)


def spin_words(sentence):
   words = [word for word in sentence.split(" ")]
   words = [word if len(word) < 5 else word[::-1] for word in words]
   return " ".join(words)


print(spin_words("Welcome"), "emocleW")
print(spin_words("Hey fellow warriors"), "Hey wollef sroirraw")
print(spin_words("This is a test"), "This is a test")
print(spin_words("This is another test"), "This is rehtona test")
print(spin_words("You are almost to the last test"), "You are tsomla to the last test")
print(spin_words("Just kidding there is still one more"), "Just gniddik ereht is llits one more")
print(spin_words("Seriously this is the last one"), "ylsuoireS this is the last one")